tests/test_adapter.py

```diff
@@ -324,3 +325,26 @@ def forward_pre_hook(module, input):
 
         self.assertEqual((1, 128, model.config.decoder.vocab_size), out[0].shape)
         self.assertEqual(2, calls)
+
+
+@require_torch
+class T5AdapterTest(
+    AdapterModelTestMixin,
+    AdapterFusionModelTestMixin,
+    PredictionHeadModelTestMixin,
+    AdapterTestBase,
+    unittest.TestCase,
+):
```

Comment:
It seems like T5 is a "full" adapter implementation (w. flex heads, parallel block, ...), so, compared to other models such as BERT & GPT-2, some additional test mixins might be useful:
- AdapterTrainingTestMixin
- ParallelAdapterInferenceTestMixin
- ParallelTrainingMixin